diffkemp/config.py

```diff
@@ -41,3 +61,60 @@ def __init__(self, snapshot_first, snapshot_second, show_diff,
                                        with -DBUILD_LLREVE=ON")
         elif semdiff_tool is not None:
             raise ConfigException("Unsupported semantic diff tool")
+
+    @classmethod
+    def from_args(cls, args):
+        """
+        Create the configuration from command line arguments.
+        :param args: Command line arguments
+        """
+        # Parse both the new and the old snapshot.
+        snapshot_first = Snapshot.load_from_dir(args.snapshot_dir_old)
+        snapshot_second = Snapshot.load_from_dir(args.snapshot_dir_new)
+
+        # Check if snapshot LLVM versions are compatible with
+        # the current version.
+        llvm_version = get_llvm_version()
+
+        if llvm_version != snapshot_first.llvm_version:
+            sys.stderr.write(
+                "Error: old snapshot was built with LLVM {}, "
+                "current version is LLVM {}.\n".format(
+                    snapshot_first.llvm_version, llvm_version
+                )
+            )
+            sys.exit(errno.EINVAL)
+
+        if llvm_version != snapshot_second.llvm_version:
+            sys.stderr.write(
+                "Error: new snapshot was built with LLVM {}, "
+                "current version is LLVM {}.\n".format(
+                    snapshot_second.llvm_version, llvm_version
+                )
+            )
+            sys.exit(errno.EINVAL)
+
+        if args.function:
+            snapshot_first.filter([args.function])
+            snapshot_second.filter([args.function])
+
+        # Transform difference pattern files into an LLVM IR
+        # based configuration.
+        if args.patterns:
+            pattern_config = PatternConfig.create_from_file(args.patterns)
+        else:
+            pattern_config = None
+
+        return cls(
+            snapshot_first=snapshot_first,
+            snapshot_second=snapshot_second,
+            show_diff=not args.no_show_diff or args.show_diff,
+            full_diff=args.full_diff,
+            pattern_config=pattern_config,
+            control_flow_only=args.control_flow_only,
+            output_llvm_ir=args.output_llvm_ir,
+            print_asm_diffs=args.print_asm_diffs,
+            verbosity=args.verbosity,
+            use_ffi=args.use_ffi,
```

Comment:
This is now a different CLI option
```suggestion
            use_ffi=not args.disable_simpll_ffi,
```